.github/workflows/release.yml

```diff
@@ -0,0 +1,25 @@
+name: "Build and attach binary to release"
+on:
+  release:
+    types: ["created"]
```

Comment:
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

FTR the types are

- published
- unpublished
- created
- edited
- deleted
- prereleased
- released